app_dial: Fix progress timeout calculation with no answer timeout.

InterLinked1 opened the pull request:
If to_answer is -1, simply comparing to see if the progress timeout is smaller than the answer timeout to prefer it will fail. Add an additional check that chooses the progress timeout if there is no answer timeout (or as before, if the progress timeout is smaller).

Resolves: #821

InterLinked1 commented:
cherry-pick-to: 18
cherry-pick-to: 20
cherry-pick-to: 21
cherry-pick-to: 22
testsuite-test-pr: 69

gtjoseph commented:
rechecking to pick up app_dial tests.

gtjoseph commented:
Still failing...

https://github.com/asterisk/asterisk/actions/runs/11375404767/job/31646098479?pr=949

InterLinked1 commented:
> Still failing...
> 
> https://github.com/asterisk/asterisk/actions/runs/11375404767/job/31646098479?pr=949

Yup :)
That's an issue with the test itself though, I added a comment there for you.

This change should be correct and when I modified the test to remove the issue with it, it does pass.

gtjoseph reviewed (approved):
This change plus the corresponding testsuite PR now passes.
